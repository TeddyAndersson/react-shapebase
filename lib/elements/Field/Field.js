import React from 'react';
import _ from 'lodash';
import PropTypes from 'prop-types';
import styled, {css} from 'styled-components';
import RectangleShape from '../Shape/RectangleShape'
import FieldLegend from './FieldLegend';

const Element = props => {
    const {
        active,
        children,
        className,
        disabled,
        error,
        ...other
    } = props;

    return (
        <fieldset className={className} {...other}>
            {children}
        </fieldset>
    );
}

Element.propTypes = {
    /**
      * Defines if the field is currently active.
      */
    active: PropTypes.bool,

    /**
      * Primary content.
      */
    children: PropTypes.node,

    /**
     * Additonal classes.
     */
    className: PropTypes.string,

    /**
     * Error classes.
     */
    error: PropTypes.bool,

    /**
     * Disabled classes.
     */
    disabled: PropTypes.bool,
}

Element.defaultProps = {
    type: 'text',
}

/** Styling the Component with styled-components */
const Field = styled(Element)`
    border: 0;
    padding: 0;
    margin: 0;
    min-width: 0;
    padding: 6% 2%;
    width: 96%;

    ${ props => props.error && css`
        color: red;
        background: lightred;
    `}
`;

Field.Legend = FieldLegend

export default Field;