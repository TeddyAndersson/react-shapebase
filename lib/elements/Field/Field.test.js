import React from 'react';
import Field from './Field'
import renderer from 'react-test-renderer';

test('should render Field', () => {
  const component = renderer.create(
    <Field>
      <Field.Legend content="A field with text input"/>
    </Field>
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})
