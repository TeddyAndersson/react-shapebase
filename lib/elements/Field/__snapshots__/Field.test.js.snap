// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Field 1`] = `
<fieldset
  className="sc-bxivhb giFaBF"
  type="text"
>
  <p
    className="sc-htpNat hxUdsp"
  >
    A field with text input
  </p>
</fieldset>
`;
